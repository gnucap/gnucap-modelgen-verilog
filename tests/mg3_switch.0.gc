

attach ../modelgen_0.so


verilog

`modelgen
module test_sw0(p, n)
	electrical p, n;
	(* desc="" *) real st;

	analog begin
		if (V(p, n) > 0) begin
			st = 1;
			I(p, n) <+ V(p,n);
		end else
			st = 0;
	end

endmodule

!make test_sw0.so > /dev/null
attach ./test_sw0.so

test_sw0 #() dut(1,0);
parameter v=0
vsource #(.dc(v)) v1(1,0);

list

print dc v(nodes) i(v1) st(dut) iter(0)
dc v1 -3 3 1 loop
end
