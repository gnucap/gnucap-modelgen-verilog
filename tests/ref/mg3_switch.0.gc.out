test_sw0 #() dut (.p(1),.n(0));
parameter v=0 
vsource #(.dc(v)) v1 (.p(1),.n(0));
#           v(1)       i(v1)      st(dut)    iter(0)   
-3.        -3.         0.         0.         3.        
-2.        -2.         0.         0.         3.        
-1.        -1.         0.         0.         3.        
 0.         0.         0.         0.         3.        
 1.         0.99999   -0.99999    1.         4.        
 2.         2.        -2.         1.         3.        
 3.         3.        -3.         1.         3.        
 2.         2.        -2.         1.         3.        
 1.         0.99999   -0.99999    1.         3.        
 0.         0.         0.         0.         4.        
-1.        -1.         0.         0.         3.        
-2.        -2.         0.         0.         3.        
-3.        -3.         0.         0.         3.        
